eng/scripts/Fetch-Namespaces-DotnetNupkg.ps1

```diff
@@ -0,0 +1,42 @@
+<#
+  .SYNOPSIS
+  Fetch the namespaces from dotnet nupkg and checkin to docs/metadata
+
+  .DESCRIPTION
+  The scripts is used for docs.ms to fetch namespace list for particular nupkg artifact.
+
+  .PARAMETER DotnetNupkgLocation
+  Specifies the fo nupkg location.
+
+  .PARAMETER DocRepoLocation
+  Specifies Location of the root of the docs.microsoft.com reference doc location. 
+
+  .PARAMETER ArtifactName
+  The artifact name. E.g. Azure.Template
+#>
+[CmdletBinding()]
+param (
+  [string] $DotnetNupkgLocation = "",
+  [string] $DocRepoLocation = "",
+  [string] $ArtifactName = ""
+)
+Set-StrictMode -Version 3
+. (Join-Path $PSScriptRoot ../common/scripts/common.ps1)
+
+Write-Host "DotnetNupkgLocation: $DotnetNupkgLocation/$ArtifactName"
+Write-Host "DocRepoLocation: $DocRepoLocation"
+
+$nupkgFile = Get-dotnet-Package-Artifacts -Location "$DotnetNupkgLocation/$ArtifactName"
+$version = $nupkgFile.Name -replace "$ArtifactName.(.*\d).nupkg", '$1'
```

Comment:
Tip: by default, use single quotes for string literals. Especially in regex, if you use a `$` (in regex, end of line) that will be interpreted as the start of a variable. Using single quotes avoids that.

Reply:
Just did a local test on single quotes. Single quotes don't return the right value. 
```
"Azure.Template.1.0.0-beta.1.nupkg" -replace "$ArtifactName.(.*\d).nupkg", '$1'
```
returns
```
1.0.0-beta.1
```

However, 
```
"Azure.Template.1.0.0-beta.1.nupkg" -replace '$ArtifactName.(.*\d).nupkg', '$1'
```
returns me
```
Azure.Template.1.0.0-beta.1.nupkg
```

Reply:
Or, you do have `$AzureName`. Sorry, I was paying too much attention to the coloring. Seems it's *not* colored in double quotes if it's a variable. Weird choice for the syntax highlighter. You're right to use double quotes here then. Sorry. But, this actually highlights a potential issue. The ArtifactName will probably be safe, but you might actually want to use something like `[Regex]::Escape()` or something like that (look in the .NET docs for what the static method is called). That said, I believe ArtifactName is typically safe. That's the one without `.`, right? Since a `.` would match any character. Even then, that's probably still safe, but may as well be 100% sure and encoded this since you're passing it to a regex input.

Reply:
It is better to match `.` literally, instead of any characters. I think we can use backslash for the purpose.

Reply:
Right, that's why you should use the static method on `[Regex]` to escape the input i.e., `$ArtifactName`. This is common for user-provided regex.

Reply:
Done. Thanks for the suggestion!